.github/workflows/test-push.yml

```diff
@@ -20,29 +20,58 @@ concurrency:
 jobs:
   build:
     runs-on: ubuntu-24.04
+    env:
+      GRADLE_OPTS: -Dorg.gradle.daemon=false -Xmx4g -XX:+UseParallelGC
     steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - uses: gradle/actions/wrapper-validation@v4
       - name: setup jdk
         uses: actions/setup-java@v4
         with:
           java-version: 21
-          distribution: 'temurin'
-      - uses: actions/checkout@v4
-      - uses: gradle/actions/wrapper-validation@v4
-      - run: ./gradlew build check -x test --stacktrace --warning-mode fail
+          distribution: 'graalvm'
+          cache: 'gradle'
+      - uses: gradle/actions/setup-gradle@v4
+        with:
+          cache-read-only: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository }}
+          gradle-home-cache-cleanup: true
+      - run: ./gradlew build check -x test --stacktrace --warning-mode fail --parallel --configuration-cache-problems=warn --build-cache --no-watch-fs
+      - uses: actions/upload-artifact@v4
+        with:
+          name: build-artifacts
+          path: |
+            build/libs/
+            build/classes/
+            */build/libs/
+            */build/classes/
+            .gradle/configuration-cache/
```

Comment:
No!!! Even if loom was using the configuration cache it should never be shared, it likely includes all secrects on the machine (even the short lived github token used for the build). Gradle encrypts this by default to prevent misuse like this.

Reply:
See: https://nvd.nist.gov/vuln/detail/CVE-2023-30853 you are opt-ing back into this vulnerability. This is fatal... if you do this on any other project consider it compromised.

I dont know where gradle caches the encryption key for the config cache, but assuming you have it working the key must be in the cache.